modules/core/test/test_mat.cpp

```diff
@@ -1248,6 +1248,46 @@ static void check_ndim_shape(const cv::Mat &mat, int cn, int ndims, const int *s
         EXPECT_EQ(mat.size[i], sizes[i]);
 }
 
+TEST(Core_Mat, convertTo)
+{
+    {
+        cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
+        void* p = A.data;
+        int cn = CV_MAT_CN(CV_8UC3);
+        int depth = CV_8S;
+        int dtype = CV_MAKE_TYPE(depth, cn);
+        A.convertTo(A, depth);
+        EXPECT_EQ(p, A.data);
+        EXPECT_EQ(A.type(), dtype);
+    }
+
+    {
+        cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
+        void* p = A.data;
+        int cn = CV_MAT_CN(CV_8UC3);
+        int depth = CV_32F;
+        int dtype = CV_MAKE_TYPE(depth, cn);
+        A.convertTo(A, depth);
+        EXPECT_NE(p, A.data);
+        EXPECT_EQ(A.type(), dtype);
+    }
+
+    {
+        cv::Mat A(8, 16, CV_8UC3, cv::Scalar(1, 2, 3));
+        void* p = A.data;
+        int cn = CV_MAT_CN(CV_8UC3);
+        int depth = CV_8S;
+        int stype = A.type();
+        int dtype = CV_MAKE_TYPE(depth, cn);
+        cv::Mat B = A;
+        A.convertTo(A, depth);
+        EXPECT_EQ(A.type(), dtype);
+        EXPECT_EQ(B.type(), stype);
+        EXPECT_NE(p, A.data);
+        EXPECT_EQ(p, B.data);
+    }
+}
+
```

Comment:
Can you please also add a test for `CV_8UC4->CV_32FC1` inplace conversion expecting that it will keep memory?

Reply:
Yes, I can. But **convertTo** only supports cases where the number of channels in the (src) and (dst) are the same. **cvtColor** support CV_8UC4->CV_32FC1. I think that **cvtColor** can also be done in-place.

Reply:
Yes, I can. But convertTo only supports cases where the number of channels in the (src) and (dst) are the same. 
Currently, I'm unable to think of any function supporting the conversion from CV_8UC4 to CV_32FC1.

Reply:
Ok, I got it, thanks!